src/server.rs

```diff
@@ -78,7 +78,7 @@ pub fn start_server(listen_at: &str, queue_size: usize, threads: u64) {
         thread::spawn(move || loop {
             let socket = rec.lock().unwrap().recv().unwrap();
             if let Err(e) = handle_connection(e.deref(), socket) {
-                println!("Error handling connection: {e}");
+                println!("Error handling connection: {}", e);
```

Comment:
Wait what is this? I would have expected the opposite as that is the one recommended by `cargo clippy`.